backend/server/src/services/matchService.ts

```diff
@@ -194,9 +196,75 @@ export class MatchService {
     if (player1Points >= MAXIMUM_POINTS) {
       match.winner = player1.id;
       match.endTimestamp = new Date();
+      await this.createPlayoffSchedule(match.id, player1.id);
     } else if (player2Points >= MAXIMUM_POINTS) {
       match.winner = player2.id;
       match.endTimestamp = new Date();
+      await this.createPlayoffSchedule(match.id, player2.id);
+    }
+  }
+
+  private async createPlayoffSchedule(matchId: Types.ObjectId, winnerId: Types.ObjectId): Promise<void> {
+    const tournament = await TournamentModel.findOne({
+      matchSchedule: matchId
+    }).populate({
+      path: 'matchSchedule',
+      model: 'Match'
+    }).exec();
+
+    if (tournament?.tournamentType !== TournamentType.Playoff) {
+      return;
+    }
+
+    const playedMatches = tournament.matchSchedule;
+
+    // Find the current round from the match
+    const currentMatch = playedMatches.find(match => match._id.equals(matchId));
+    if (!currentMatch) {
+      throw new NotFoundError({
+        message: "Match not found in tournament schedule"
+      });
+    }
+    const currentRound = currentMatch.tournamentRound;
+    const nextRound = currentRound + 1;
+
+    // Filter for winners in the current round
+    const winners = playedMatches
+      .filter(match => match.tournamentRound === currentRound && match.winner)
+      .map(match => match.winner);
+
+    // Find eligible winners who don't have a match in the next round
+    const eligibleWinners = winners.filter(winner => {
+      return !playedMatches.some(match =>
+        match.tournamentRound === nextRound &&
+        match.players.some(player => player.id.toString() == winner.toString())
+      );
+    });
+
+    // Pair current winner with eligible winners for the next round
+    for (const pairWithWinnerId of eligibleWinners) {
+      if (!pairWithWinnerId.equals(winnerId)) {
+        // Create a new match
+        const newMatch = {
+          players: [
+            { id: winnerId, points: [], color: "red" },
+            { id: pairWithWinnerId, points: [], color: "white" }
+          ],
+          type: "playoff",
+          admin: null,
+          elapsedTime: 0,
+          timerStartedTimestamp: null,
+          tournamentRound: nextRound
+        };
+
+        const matchDocuments = await MatchModel.create(newMatch);
+        tournament.matchSchedule.push(matchDocuments.id);
+      }
+    }
```

Comment:
The DB operation should be moved outside of the loop.

Reply:
No. Its just one operations. There would be only two eligibleWinner, one is the current winnerID and another is the one winnerID will be paired with. There's no possibility of ever having 3 winners in a playoff match who doesn't have a match in their respected round